### PR TITLE
Adjust implementation of table functions

### PR DESCRIPTION
**Pull request by 4z0t**
* `table.getsize` used to return 0 when given non table value. Make asm implementation do the same as well. related to https://github.com/FAForever/FA-Binary-Patches/issues/97;
* implement `table.getsize` in C++;
* implement `table.empty` in C++.
* don't use asm to register new functions

### CODE REVIEW

**Comment by 4z0t** [comment]
Made it compatible with Python patcher

File: section/Lua/TableFuncs.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    for (int i = start_i; i <= end_i; i++)
+    {
+        lua_rawgeti(l, 1, i);
+    }
+    return n_stack;
+}
+
+const luaL_reg RegTableFuncsDesc[] = {{"getsize2", &lua_tablesize},
+                                      {"empty2", &lua_tableempty},
+                                      {"getn2", (lua_CFunction)0x00927C20},
```
Comment (Garanas) [inline section/Lua/TableFuncs.cpp:165]: Do we still need these `2` alternatives?

Comment (4z0t) [reply section/Lua/TableFuncs.cpp:165]: I can remove them.

Comment (4z0t) [reply section/Lua/TableFuncs.cpp:165]: The problem is Lua resets them in current dev branch:
```lua
--- table.empty(t) returns true iff t has no keys/values.
---@param t table
---@return boolean
function table.empty(t)
    if type(t) ~= 'table' then return true end
    return next(t) == nil
end

--- Returns actual size of a table, including string keys
---@param t table
---@return number
function table.getsize(t)
    if type(t) ~= 'table' then return 0 end
    local size = 0
    for k, v in t do
        size = size + 1
    end
    return size
end

-- replace with assembly implementations
table.empty = table.empty2 or table.empty
table.getsize = table.getsize2 or table.getsize

```

Comment (Garanas) [reply section/Lua/TableFuncs.cpp:165]: Then we remove that logic. As long as the assembly versions are functionally equivalent then that should be no issue.

Comment (Garanas) [reply section/Lua/TableFuncs.cpp:165]: See also: https://github.com/FAForever/fa/pull/6537

Comment (4z0t) [reply section/Lua/TableFuncs.cpp]: Done

**Comment by 4z0t** [comment]
If we want to have deepcopy here is how it was done in my dev branch https://github.com/4z0t/FA-Binary-Patches/blob/develop/section/Lua/LuaObject.cxx#L87-L121

**Pull request merged**